docs/okeps/okep-5094-layer2-transit-router.md

```diff
@@ -0,0 +1,870 @@
+
+For IPv6 two main things need to be done
+- Send a unsolicited RA to configure the ovn_cluster_router gateway
+- Send one unsolicited RA per node with lifetime 0 to remove the gateways from GRs.
+
+#### Upgrade cleanups
+
+The implementation should handle cleaning up the ports, routes, policies and NAT that are no longer
+configured on the gateway router or layer2 switch and remove them so there is no stale 
+configuration from the previous.
+
+### Rolling upgrade and traffic disruption
+
+During the rolling upgrades the nodes are upgrades one after another so cluster
+is transitory with some nodes with previous non transit router topology and
+others with the new transit router topology.
+
+To prevent traffic disruption at this scenario the remote connection between
+nodes has to be kept, following is a possible transitory topology to fix this
+issue:
+
+```mermaid
+%%{init: {"nodeSpacing": 20, "rankSpacing": 150}}%%
+flowchart TD
+    classDef nodeStyle fill:orange,stroke:none,rx:10px,ry:10px,font-size:25px;
+    classDef vmStyle fill:blue,stroke:none,color:white,rx:10px,ry:10px,font-size:25px;
+    classDef portStyle fill:#3CB371,color:black,stroke:none,rx:10px,ry:10px,font-size:25px;
+    classDef routerStyle fill:brown,color:white,stroke:none,rx:10px,ry:10px,font-size:25px;
+    classDef switchStyle fill:brown,color:white,stroke:none,rx:10px,ry:10px,font-size:25px;
+    classDef termStyle font-family:monospace,fill:black,stroke:none,color:white;
+    
+    subgraph node1["node1 <b>(updated)</b>"]
+        subgraph GR-node1
+            rtotr-GR-node1["trtor-GR-node1 
+            100.65.0.2/16 100.88.0.6/30 (0a:58:64:41:00:02)"]
+        end
+        
+        subgraph layer2-switch["layer2-switch"]
+            stor-ovn_cluster_router["stor-ovn_cluster_router 
+                type: router"]
+            stor-GR-node3["stor-GR-node3  100.65.0.4/16 (0a:58:64:41:00:04)
+                type: remote"]
+        end
+        
+        
+        
+        subgraph ovn_cluster_router["ovn_cluster_router "]
+            trtor-GR-node2["trtor-GR-node2 100.88.0.13/30"]
+            trtor-GR-node1["trtor-GR-node1 100.88.0.5/30"]
+            rtos-layer2-switch["rtos-layer2-switch 203.203.0.1/24 <b>100.65.0.2/16</b> (0a:58:CB:CB:00:01)"]
+        end
+    end
+    
+    subgraph node2["node2 <b>(updated)</b>"]
+        subgraph GR-node2
+            rtotr-GR-node2["rtotr-GR-node2 100.65.0.3/16 100.88.0.14/30 (0a:58:64:41:00:03)"]
+        end
+    end
+    
+    subgraph node3["node3 <b>(old topo)</b>"]
+        subgraph VM["Virtual Machine"]
+            class VM vmStyle;
+            term["default gw 
+            203.203.0.1
+            (0a:58:64:41:00:04)"]
+        end
+        subgraph GR-node3
+            rtos-GR-node3["rtos-GR-node3  203.203.0.1/24 100.65.0.4/16 (0a:58:64:41:00:04)"]
+        end
+        subgraph layer2-switch-node3["layer2-switch"]
+            stor-GR-node1["stor-GR-node1
```

Comment:
why does the non-updated node3 have a port to stor-GR-node1 on it with a colliding 100.65.0.2 address? This should be removed from node3's switch when node 1 updated.